rocAL/rocAL_pybind/amd/rocal/fn.py

```diff
@@ -29,219 +29,437 @@
 import rocal_pybind as b
 from amd.rocal.pipeline import Pipeline
 
-def blend(*inputs,ratio=None):
-    kwargs_pybind = {"input_image0":inputs[0], "input_image1":inputs[1], "is_output":False ,"ratio":ratio}
-    blend_image = b.Blend(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+
+def blend(*inputs, ratio=None, rocal_tensor_output_layout=types.NHWC, rocal_tensor_output_datatype=types.UINT8):
+    """
+    inputs - list containing the input images
+    
+    ratio (float, optional, default = None) - ratio used for blending one image with another
+
+    rocal_tensor_output_layout (int, optional, default = types.NHWC) - tensor layout for the augmentation output
+
+    rocal_tensor_output_datatype (int, optional, default = types.UINT8) - tensor dtype for the augmentation output
+    """
+    ratio = b.createFloatParameter(ratio) if isinstance(ratio, float) else ratio
+    # pybind call arguments
+    kwargs_pybind = {"input_image0": inputs[0], "input_image1": inputs[1], "is_output": False, "ratio": ratio,
+                     "rocal_tensor_output_layout": rocal_tensor_output_layout, "rocal_tensor_output_datatype": rocal_tensor_output_datatype}
+    blend_image = b.Blend(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (blend_image)
 
-def snow(*inputs, snow=0.5, device=None):
+
+def snow(*inputs, snow=0.5, rocal_tensor_output_layout=types.NHWC, rocal_tensor_output_datatype=types.UINT8):
+    """
+    inputs - the input image passed to the augmentation
+    
+    snow (float, default = 0.5) - snow fill value used for the augmentation
+
+    rocal_tensor_output_layout (int, optional, default = types.NHWC) - tensor layout for the augmentation output
+
+    rocal_tensor_output_datatype (int, optional, default = types.UINT8) - tensor dtype for the augmentation output
+    """
+    snow = b.createFloatParameter(snow) if isinstance(snow, float) else snow
     # pybind call arguments
-    snow = b.CreateFloatParameter(snow) if isinstance(snow, float) else snow
-    kwargs_pybind = {"input_image0": inputs[0],"is_output": False, "shift": snow}
-    snow_image = b.Snow(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    kwargs_pybind = {"input_image": inputs[0], "is_output": False, "snow": snow,
+                     "rocal_tensor_output_layout": rocal_tensor_output_layout, "rocal_tensor_output_datatype": rocal_tensor_output_datatype}
+    snow_image = b.Snow(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (snow_image)
 
-def exposure(*inputs, exposure=0.5, device=None):
+
+def exposure(*inputs, exposure=0.5, rocal_tensor_output_layout=types.NHWC, rocal_tensor_output_datatype=types.UINT8):
+    """
```

Comment:
shouldn't the function descriptor be added before the function defenition?
Please check the norm.

Reply:
Its the norm in python for the docstrings to be added after the function definition